active-rfcs/0000-relative-url-scheme.md

```diff
@@ -0,0 +1,95 @@
+- Start Date: 2021-12-20
+- Reference Issues: [#46](https://github.com/withastro/rfcs/pull/46)
+- Implementation PR: <!-- leave this empty -->
+
+# Summary
+
+Astro should support source-relative URL resolution in the HTML block of Astro files.
+
+# Example
+
+To resolve the source-relative URL of an image, use a `local:` prefixed URL.
+
+```html
+<img src="local:kitten.avif" alt="Kitten" />
+```
+
+# Motivation
+
+In `.astro` files, it’s not always easy to know what a relative URL is relative to.
+
+```astro
+<img src="kitten.avif" alt="kitty" />
+```
+
+In this example, is `kitten.avif` relative to the output `.html` or the source `.astro`? Or, if this example is found within a component `.astro` file, is it relative to that component file?
+
+This ambiguity raises a need for:
+
+- An intuitive way to author URLs relative to a source file.
+- An intuitive way to author URLs within the HTML of an `.astro` file.
+
+# Detailed design
+
+A `local` scheme instructs Astro to resolve the URL relative to the source file.
+
+```astro
+<img srcset="local:kitten.avif 1x, local:kitten@2x.avif" alt="Kitten" />
+```
+
+```astro
+<picture>
+  <source srcset="local:media/kitten@2x.avif" media="(min-width: 800px)" />
+  <img src="local:media/kitten.jpg" alt="" />
+</picture>
+```
+
+```astro
+<div style="background-image: url(local:kitten-background.avif);"></div>
+```
+
+```astro
+<div data-my-framework-attr-for-image="local:kitten-background.avif"></div>
+```
+
+# Drawbacks
+
+All attribute values need to be checked for `local:`. See alternatives below
+
+The provided examples are intuitive and convenient for authors, but not immediately obvious to an implementer.
+
+# Alternatives
+
+- Limit support to `local:` prefixed attributes
+  ```html
+  <img local:src="kitten.avif" alt="Kitten" />
+  ```
+  - It’s not obvious if this would be more performant than checking every attribute for a `local:` prefix.
+  - It’s even less obvious if this would be more performant when the attribute value may include multiple URLs (the `srcset`, `style`, or `data-` attributes), or wraps the URL in `url()`, `url("")`, `url('')`, etc.
```

Comment:
Is performance the biggest issue here? I feel like complexity is a big difference between these two that I don't see mentioned here. 
- Implementation complexity: basically the convo above about how we could support this across all use-cases, when it's needed at compile time. 
- User complexity: How can we document this well to developers? ex: you couldn't do `src={someVar}` because it needs to be scanned at compile-time, right?

Reply:
Yea, the focus of this section should not have been performance, but complexity.

I’ve updated this section to focus on the complexity of a namespaced attribute versus a scheme. Would you check it out and let me know if you feel it better addresses your issues?

Reply:
This isn’t necessarily resolved by the updates in aa6df9d5fa30d5086f5562b205a2015f69939721, but it did not come up in the RFC discussion, so I am resolving this thread.